### PR TITLE
Feat: Add safe test app to the repo

### PR DESCRIPTION
**Pull request by mikhailxyz**
This PR:
- Partly implements https://github.com/safe-global/safe-react-apps/issues/473 by moving the original test app from https://github.com/gnosis/safe-test-app to the safe-apps-sdk repo. I couldn't make the vite app work in a monorepo, so I think we should either deprecate it or keep it as a separate repo.
- Updates all dependencies 💪 

How to test:
1. checkout to the branch
2. run `yarn workspace @gnosis.pm/safe-apps-test-app start`
3. Add a custom safe app to the test interface and check if it works

### CODE REVIEW

File: packages/safe-apps-test-app/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      .getUserMedia({ video: true, audio: true })
+      .then(function (stream) {
+        console.log('fine');
+      })
+      .catch(function (err) {
+        console.error(err);
+      });
+
+    async function loadSafeInfo() {
+      const safuInfo = await SDK.safe.getInfo()
```
Comment (DaniSomoza) [inline packages/safe-apps-test-app/src/App.tsx:45]: Minor typo here right? `safuInfo` -> `safeInfo `

File: package.json [hunk 2]
```diff
@@ -19,13 +19,13 @@
     "@changesets/cli": "^2.25.0",
```
Comment (DaniSomoza) [inline package.json:19]: can we add a `start` script in the main `package.json`:

```

    "start:test-appt": "yarn workspace @gnosis.pm/safe-apps-test-app start",
```

And update the `Readme.md` file

File: packages/safe-apps-test-app/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+## Running the app
+
+```
+yarn install
+yarn start
+```
+
+Then:
+
+- Go to Rinkeby version of Gnosis Safe [https://rinkeby.gnosis-safe.io/app](https://rinkeby.gnosis-safe.io/app)
```
Comment (yagopv) [inline packages/safe-apps-test-app/README.md:12]: Need to update this URL

File: packages/safe-apps-test-app/public/index.html [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    <!--
+      Notice the use of %PUBLIC_URL% in the tags above.
+      It will be replaced with the URL of the `public` folder during the build.
+      Only files inside the `public` folder can be referenced from the HTML.
+
+      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <title>Gnosis Safe App</title>
```
Comment (yagopv) [inline packages/safe-apps-test-app/public/index.html:27]: Remove Gnosis 😬

File: packages/safe-apps-sdk/dist/package.json [hunk 5]
```diff
@@ -23,7 +23,7 @@
     "author": "Gnosis (https://gnosis.io)",
```
Comment (yagopv) [inline packages/safe-apps-sdk/dist/package.json:23]: Remove Gnosis as well

File: packages/safe-apps-test-app/package.json [hunk 6]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "@gnosis.pm/safe-apps-test-app",
```
Comment (yagopv) [inline packages/safe-apps-test-app/package.json:2]: If we remove @gnosis.pm then we can `yarn workspace safe-apps-test-app start` without the prefix

File: packages/safe-apps-test-app/src/GlobalStyle.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+    body {
+       height: 100%;
+       margin: 0px;
+       padding: 0px;
+    }
+
+    #root {
+        height: 100%;
+        padding-right: 0.5rem;
```
Comment (yagopv) [inline packages/safe-apps-test-app/src/GlobalStyle.ts:16]: we can add background: white for dealing with the web-core dark mode

File: packages/safe-apps-test-app/package.json [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+  "version": "0.0.1",
+  "description": "Safe App for testing purposes",
+  "author": "Safe Foundation",
+  "license": "MIT",
+  "private": true,
+  "homepage": ".",
+  "dependencies": {
+    "@gnosis.pm/safe-apps-sdk": "^7.8.0",
```
Comment (yagopv) [inline packages/safe-apps-test-app/package.json:10]: We can link this to the local repo: 

"@gnosis.pm/safe-apps-sdk": "file:../safe-apps-sdk"

Ideally after each `build:sdk` we should copy the `dist` directory from `safe-apps-sdk` to the `safe-apps-test-app`  corresponding directory inside node_modules. This way we could make the testing faster. But anyway this is a nice to have we can implement in any moment ...

Comment (mikhailxyz) [reply packages/safe-apps-test-app/package.json:10]: https://classic.yarnpkg.com/lang/en/docs/workspaces/#toc-why-would-you-want-to-do-this

Yarn workspaces already do it

File: packages/safe-apps-test-app/src/App.tsx (hunk 1 above)
Comment (mikhailxyz) [reply packages/safe-apps-test-app/src/App.tsx:45]: Not really a typo, it comes from "funds are safu" meme

Comment (mikhailxyz) [reply packages/safe-apps-test-app/src/App.tsx:45]: https://www.urbandictionary.com/define.php?term=Safu

Comment (mikhailxyz) [reply packages/safe-apps-test-app/src/App.tsx:45]: if you find it v confusing I can change it no problem

File: package.json (hunk 2 above)
Comment (mikhailxyz) [reply package.json:20]: done senor

File: packages/safe-apps-test-app/README.md (hunk 3 above)
Comment (mikhailxyz) [reply packages/safe-apps-test-app/README.md]: done senor

File: packages/safe-apps-test-app/public/index.html (hunk 4 above)
Comment (mikhailxyz) [reply packages/safe-apps-test-app/public/index.html:27]: done senor

File: packages/safe-apps-sdk/dist/package.json (hunk 5 above)
Comment (mikhailxyz) [reply packages/safe-apps-sdk/dist/package.json:23]: done senor

File: packages/safe-apps-test-app/package.json (hunk 6 above)
Comment (mikhailxyz) [reply packages/safe-apps-test-app/package.json:2]: currently all the packages follow the same convention `@gnosis.pm/package-name` so let's keep it until we get a new npm org name

File: packages/safe-apps-test-app/src/App.tsx (hunk 1 above)
Comment (yagopv) [reply packages/safe-apps-test-app/src/App.tsx:45]: 😂 Ok I think Dani will be fine with your explanation

Comment (yagopv) [reply packages/safe-apps-test-app/src/App.tsx:45]: no idea about this

File: packages/safe-apps-test-app/package.json (hunk 6 above)
Comment (yagopv) [reply packages/safe-apps-test-app/package.json:2]: ok!

**Review by yagopv — APPROVED** [review]

**Pull request merged**